packages/@jsii/kernel/src/kernel.ts

```diff
@@ -1278,6 +1306,13 @@ export class Kernel {
     this.syncInProgress = desc;
     try {
       return fn();
+    } catch (e: any) {
+      if (e.name === JsiiErrorType.JSII_FAULT) {
+        throw new JsiiFault(e);
+      }
+      // default to JsError because any JsiiFaults have been emitted from the kernel
+      // with their type field populated before the request was made to the js code.
+      throw new JsError(e);
```

Comment:
I think we should probably default to `JsiiFault` and explicitly check for other error types as errors without a `name` set are unknown and likely not recoverable.

Reply:
discussed offline...this needs to default to JsError since the Faults will have their type specified, and JsErrors won't